File: 최윤경/가장 큰 수.js
/*
[접근]

1. 숫자를 문자열로 변환 → 이어 붙였을 때의 순서 비교
2. 두 수를 앞뒤로 붙여 비교 → (b + a) - (a + b)
3. 내림차순 정렬 후 문자열로 이어 붙임
4. 결과가 "0"으로 시작 → 모든 수가 0이므로 "0" 반환
*/

function solution(numbers) {
  var answer = numbers
    .map(String)
    .sort((a, b) => b + a - (a + b))
    .join("");

  return answer[0] === "0" ? "0" : answer;
}

File: 표/6_hash_sort/가장큰수.py
"""
0 또는 양의 정수가 주어졌을 때, 정수를 이어 붙여 만들 수 있는 가장 큰 수를 알아내 주세요.

- 모든 숫자를 같은 자릿수에서 비교
"""
def solution(numbers):
    answer = ''
    str_num = list(map(str, numbers))
    str_num.sort(key=lambda x: x*3, reverse=True)
    
    if all(s == '0' for s in str_num):
        return '0'
    
    return (answer.join(str_num))